src/main/java/com/cleanroommc/groovyscript/compat/mods/ModPropertyContainer.java

```diff
@@ -1,57 +1,114 @@
 package com.cleanroommc.groovyscript.compat.mods;
 
-import com.cleanroommc.groovyscript.api.*;
+import com.cleanroommc.groovyscript.GroovyScript;
+import com.cleanroommc.groovyscript.api.GroovyBlacklist;
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.INamed;
+import com.cleanroommc.groovyscript.sandbox.expand.ExpansionHelper;
 import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnmodifiableView;
 
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.Map;
 
-public class ModPropertyContainer implements IDynamicGroovyProperty {
+public class ModPropertyContainer {
```

Comment:
since this isnt going to be a `ModPropertyContainer` anymore (used for vanilla, multiple times per mod, etc), and significant parts of it are getting deprecated, i think the class as a whole should be deprecated and a new class created to replace it.
heres a commit of what it could look like https://github.com/CleanroomMC/GroovyScript/commit/4bdc3ad9a3424dacf99c16905efcda1338802662

Reply:
Whats the purpose? It still does the same thing. It just doesnt store the properties by itself anymore